src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -1906,15 +1906,41 @@ def dft(self, form="seminormal", mult='l2r'):
             [   0    1    0   -1    1   -1]
             [   1 -1/2    1 -1/2 -1/2 -1/2]
             [   1   -1   -1    1    1   -1]
-        """
+
+            In cases where `p \\mid n!`, we use the modular Fourier transform.
+
+            sage: GF2S3 = SymmetricGroupAlgebra(GF(2), 3)
+            sage: GF2S3.dft()
+            [1 0 0 0 1 0]
+            [0 1 0 0 0 1]
+            [0 0 1 0 0 1]
+            [0 0 0 1 1 0]
+            [1 0 0 1 1 0]
+            [0 1 1 0 0 1]
+
+            sage: GF3S3 = SymmetricGroupAlgebra(GF(3), 3)
+            sage: GF3S3.dft()
+            [1 0 0 0 0 0]
+            [0 1 0 0 0 0]
+            [0 0 1 0 0 0]
+            [0 0 0 1 0 0]
+            [0 0 0 0 1 0]
+            [0 0 0 0 0 1]
+        """
+        if form is None:
+            form = "modular" if self.base_ring().characteristic().divides(self.group().cardinality()) else "seminormal"
         if form == "seminormal":
+            if self.base_ring().characteristic().divides(self.group().cardinality()):
+                raise ValueError("seminormal does not work when p | n!")
             return self._dft_seminormal(mult=mult)
+        if form == "modular":
+            return self._dft_modular()
         else:
             raise ValueError("invalid form (= %s)" % form)
```

Comment:
```suggestion
        raise ValueError("invalid form (= %s)" % form)
```
Personal idiosyncrasy, but I find this more clear (and robust) that all other cases should raise an error.